multiplier_analysis/nbconverted/1_get_multiplier_LV_coverage.py

```diff
@@ -49,265 +51,249 @@
 
+# In[10]:
 
-multiplier_model_z_processed.head()
 
+# Drop gene ids not used in multiplier analysis
+processed_dict_genes = process.process_generic_specific_gene_lists(dict_genes, multiplier_model_z)
 
-# ## Quick looks at the data
 
 # In[11]:
 
 
-# Get a rough sense for how many genes contribute to a given LV
-# (i.e. how many genes have a value > 0 for a given LV)
-(multiplier_model_z > 0).sum().sort_values(ascending=True)
+# Check numbers add up
+assert len(shared_genes) == len(processed_dict_genes["generic"]) + len(processed_dict_genes["other"])
 
 
-# In[12]:
+# ## Get coverage of LVs
+# 
+# For each gene (generic or other) we want to find:
+# 1. The number of LVs that gene is present
+# 2. The number of LVs that the gene contributes a lot to (i.e. the gene is highly weighted within that LV)
 
+# ### Nonzero LV coverage
 
-# One off just to get a sense for how many genes are being compared
-# Filter genes to only use those shared between our analysis and multiplier
-# Check overlap between multiplier genes and our genes
-multiplier_genes = list(multiplier_model_z.index)
-our_genes = list(pd.read_csv(ls_data_files[0], sep="\t", index_col=0, header=0).index)
-shared_genes = set(our_genes).intersection(multiplier_genes)
+# In[12]:
 
-print(len(our_genes))
-print(len(shared_genes))
 
+dict_nonzero_coverage = process.get_nonzero_LV_coverage(processed_dict_genes, multiplier_model_z)
 
-# ## Gene coverage of LV
 
 # In[13]:
 
 
-generic_cov_ls = []
-specific_cov_ls = []
+# Check genes mapped correctly
+assert processed_dict_genes["generic"][0] in dict_nonzero_coverage["generic"].index
+assert len(dict_nonzero_coverage["generic"]) == len(processed_dict_genes["generic"])
+assert len(dict_nonzero_coverage["other"]) == len(processed_dict_genes["other"])
 
-for ifile in range(len(ls_data_files)):
-    generic_genes = ls_genes_out[ifile][0]
-    specific_genes = ls_genes_out[ifile][1]
-    
-    # Only include those genes that are in multiplier otherwise will get NAs
-    generic_genes_processed, specific_genes_processed = process.process_generic_specific_gene_lists(
-        generic_genes, 
-        specific_genes, 
-        multiplier_model_z_processed
-    )
-    print(len(generic_genes_processed), len(specific_genes_processed))
-    
-    generic_cov_i, specific_cov_i = process.get_LV_coverage(
-        generic_genes_processed,
-        specific_genes_processed,
-        multiplier_model_z_processed
-    )
-    
-    generic_cov_ls += list(generic_cov_i)
-    specific_cov_ls += list(specific_cov_i)
-    
-from matplotlib_venn import venn2
-venn2([set(generic_cov_ls), set(specific_cov_ls)],
-      set_labels=("Generic LVs", "Specific LVs")
-     )
 
+# ### High weight LV coverage
 
 # In[14]:
 
 
-# Create table of unique generic LVs, unique specific LVs, shared LVs
-process.create_LV_df(generic_cov_ls, specific_cov_ls, multiplier_model_summary)
+# Quick look at the distribution of gene weights per LV
+sns.distplot(multiplier_model_z["LV2"])
```

Comment:
You can probably add `kde=False` here, I don't think the KDE is really adding much